Warnings with GCC7

When compiling d074ccdb with GCC7 following warnings are raised:

```
/home/janisozaur/workspace/falltergeist/src/Format/Acm/Unpacker.cpp: In member function ‘int Falltergeist::Format::Acm::ValueUnpacker::linearFill(int, int)’:
/home/janisozaur/workspace/falltergeist/src/Format/Acm/Unpacker.cpp:206:53: warning: left shift of negative value [-Wshift-negative-value]
     short *lb_ptr = _buffMiddle + ((-1l) << (ind - 1));
                                                     ^
/home/janisozaur/workspace/falltergeist/src/VM/Handler/Opcode8039Handler.cpp: In member function ‘virtual void Falltergeist::VM::Handler::Opcode8039::_run()’:
/home/janisozaur/workspace/falltergeist/src/VM/Handler/Opcode8039Handler.cpp:91:39: warning: this statement may fall through [-Wimplicit-fallthrough=]
                                 _error("op_add - FLOAT+STRING not allowed");
                                 ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/janisozaur/workspace/falltergeist/src/VM/Handler/Opcode8039Handler.cpp:93:29: note: here
                             case StackValue::Type::INTEGER: // INTEGER + STRING
                             ^~~~
/home/janisozaur/workspace/falltergeist/src/VM/Handler/Opcode8039Handler.cpp:95:39: warning: this statement may fall through [-Wimplicit-fallthrough=]
                                 _error("op_add - INTEGER+STRING not allowed");
                                 ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/janisozaur/workspace/falltergeist/src/VM/Handler/Opcode8039Handler.cpp:97:29: note: here
                             default:
                             ^~~~~~~
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,7 @@
 - [bugfix][#468] Check for animation if it actually has frm file (alexeevdv)
 - Sprites rendering improvement (Zervox)
 - [feature][#444] CritterAnimationHelper added. Code refactoring (alexeevdv)(dreamer-dead)
+- [bugfix][#532] Missing break statements are added in Opcode8039 switch cases (alexeevdv)
 
 0.3.0 (2016-03-16)
 =======================

--- a/src/Format/Acm/Unpacker.cpp
+++ b/src/Format/Acm/Unpacker.cpp
@@ -204,10 +204,11 @@ int ValueUnpacker::zeroFill(int pass, int /*ind*/)
 int ValueUnpacker::linearFill(int pass, int ind)
 {
     int mask = (1 << ind) - 1;
-    short *lb_ptr = _buffMiddle + ((-1l) << (ind - 1));
+    short *lb_ptr = _buffMiddle + ((-1l) << (unsigned)(ind - 1));
 
-    for (int i = 0; i < _subblocks; i++)
+    for (int i = 0; i < _subblocks; i++) {
         _blockPtr[i * _sbSize + pass] = lb_ptr[_getBits(ind) & mask];
+    }
     return 1;
 }
 

--- a/src/VM/Handler/Opcode8039Handler.cpp
+++ b/src/VM/Handler/Opcode8039Handler.cpp
@@ -89,10 +89,12 @@ namespace Falltergeist
                             case StackValue::Type::FLOAT: // FLOAT + STRING
                             {
                                 _error("op_add - FLOAT+STRING not allowed");
+                                break;
                             }
                             case StackValue::Type::INTEGER: // INTEGER + STRING
                             {
                                 _error("op_add - INTEGER+STRING not allowed");
+                                break;
                             }
                             default:
                             {

